server/routes/experiments/biomed_nl/traversal.py

```diff
@@ -686,21 +693,32 @@ def find_start_and_traversal_type(self):
       str_to_raw[entity.sanitized_str] = entity.raw_str
       for synonym in entity.synonyms:
         str_to_raw[synonym] = entity.raw_str
-    entities_to_dcids, entities_to_recognized_types = recognize_entities_from_query(
-        ' '.join((str_to_raw.keys())))
+
+    detected_entities = recognize_entities_from_query(' '.join(
+        (str_to_raw.keys())))
+
+    displayed_entities = {}
 
     start_dcids = set()
-    # Add dcids of only the start entity to start_dcids.
-    for entity, dcids in entities_to_dcids.items():
-      if entity in start_strs:
-        start_dcids.update(dcids)
     raw_to_types = {}
-    for entity, types in entities_to_recognized_types.items():
-      raw_to_types.setdefault(str_to_raw[entity], []).extend(types)
+    for entity in detected_entities:
+      raw_to_types.setdefault(str_to_raw.get(entity.name, entity.name),
+                              []).extend(entity.types)
+      if any(entity.name in start_str for start_str in start_strs):
```

Comment:
slightly confused if entity.name is actual name of the entity as it is in the KG or the string in the query that resolved to this entity? Maybe add some comments clarifying this

Reply:
This is the name as it is in the KG. Added a comment!